packages/php-wasm/universal/src/lib/base-php.ts

```diff
@@ -507,14 +508,21 @@ export abstract class BasePHP implements IsomorphicLocalPHP {
 
 	/** @inheritDoc */
 	@rethrowFileSystemError('Could not list files in "{path}"')
-	listFiles(path: string): string[] {
+	listFiles(
+		path: string,
+		options: ListFilesOptions = { prependPath: false }
+	): string[] {
 		if (!this.fileExists(path)) {
 			return [];
 		}
 		try {
-			return this[__private__dont__use].FS.readdir(path).filter(
+			const files = this[__private__dont__use].FS.readdir(path).filter(
 				(name: string) => name !== '.' && name !== '..'
 			);
+			if (options.prependPath) {
+				return files.map((name: string) => `${path}/${name}`);
```

Comment:
nit: `path` could contain a trailing slash – let's make sure this doesn't insert a double slash like `//`

Reply:
OK, now it strips any trailing slash from the given path before prepending it.

```ts
const prepend = path.replace(/\/$/, '');
return files.map((name: string) => `${prepend}/${name}`);
```